File: Java Barcode Scanner ME/app/src/main/java/de/dimedis/mobileentry/backend/CommonArg.java
package de.dimedis.mobileentry.backend;

import android.content.Intent;

import de.dimedis.mobileentry.ConfigPref;
import de.dimedis.mobileentry.MobileEntryApplication;
import de.dimedis.mobileentry.util.ConfigPrefHelper;

public class CommonArg extends BaseArg implements CommonArgImpl {
    private String userSession;
    private String userSuid;
    private String userName;
    private String fair;
    private String border;

    public CommonArg() {
    }

    public CommonArg(Intent intent) {
        super(intent);
        this.intent = intent;
    }

    @Override
    public CommonArg getParamFromIntent() {
        super.getParamFromIntent();
        setUserSession(intent.getStringExtra(BackendService.EXTRA_USER_SESSION));
        setUserSuid(intent.getStringExtra(BackendService.EXTRA_USER_SUID));
        setUserName(intent.getStringExtra(BackendService.EXTRA_USER_NAME));
        setFair(intent.getStringExtra(BackendService.EXTRA_FAIR));
        setBorder(intent.getStringExtra(BackendService.EXTRA_BORDER));
        return this;
    }

    public static void setArgToIntent(Intent intent) {

        ConfigPref pref = MobileEntryApplication.getConfigPreferences();
        BaseArg.setArgToIntent(intent);

        intent.putExtra(BackendService.EXTRA_USER_SESSION, pref.userSession());
        intent.putExtra(BackendService.EXTRA_USER_SUID, pref.userSuid());
        intent.putExtra(BackendService.EXTRA_USER_NAME, pref.userName());

        if (null != ConfigPrefHelper.getUsersBorder()) {
            intent.putExtra(BackendService.EXTRA_FAIR, ConfigPrefHelper.getUsersBorder().getFair());// fair
            intent.putExtra(BackendService.EXTRA_BORDER, ConfigPrefHelper.getUsersBorder().getBorder());//border
        } else {
            intent.putExtra(BackendService.EXTRA_FAIR, "");// fair
            intent.putExtra(BackendService.EXTRA_BORDER, "");//border
        }
    }

    public static CommonArg fromPreferences() {
        CommonArg commonArg = new CommonArg();
        ConfigPref pref = MobileEntryApplication.getConfigPreferences();

        String lang = pref.currentLanguage();
        String commKey = pref.commKey();
        String deviceSuid = pref.deviceSuid();
        String deviceType = pref.deviceType();
        String userSession = pref.userSession();
        String userSuid = pref.userSuid();
        String userName = pref.userName();
        String fair = ConfigPrefHelper.getUsersBorder().getFair();
        String border = ConfigPrefHelper.getUsersBorder().getBorder();

        commonArg.setLang(lang);
        commonArg.setCommKey(commKey);
        commonArg.setDeviceSuid(deviceSuid);
        commonArg.setDeviceType(deviceType);
        commonArg.setUserSession(userSession);
        commonArg.setUserSuid(userSuid);
        commonArg.setUserName(userName);
        commonArg.setFair(fair);
        commonArg.setBorder(border);

        return commonArg;
    }

    public String getUserSession() {
        return userSession;
    }

    public void setUserSession(String userSession) {
        this.userSession = userSession;
    }

    public String getUserSuid() {
        return userSuid;
    }

    public void setUserSuid(String userSuid) {
        this.userSuid = userSuid;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String getFair() {
        return fair;
    }

    public void setFair(String fair) {
        this.fair = fair;
    }

    public String getBorder() {
        return border;
    }

    public void setBorder(String border) {
        this.border = border;
    }
}

File: Kotlin Migration/NTMobileEntry/app/src/main/java/de/dimedis/mobileentry/backend/CommonArg.kt
package de.dimedis.mobileentry.backend

import android.content.Intent
import de.dimedis.mobileentry.ConfigPref
import de.dimedis.mobileentry.util.ConfigPrefHelper

class CommonArg(var intent: Intent?) : BaseArg(intent) {
    var userSession: String? = null
    var userSuid: String? = null
    var userName: String? = null
    var fair: String? = null
    var border: String? = null

    override fun getParamFromIntent(): CommonArg {
        super.getParamFromIntent()
        userSession(intent?.getStringExtra(BackendServiceConst.EXTRA_USER_SESSION))
        userSuid(intent?.getStringExtra(BackendServiceConst.EXTRA_USER_SUID))
        userName(intent?.getStringExtra(BackendServiceConst.EXTRA_USER_NAME))
        fair(intent?.getStringExtra(BackendServiceConst.EXTRA_FAIR))
        border(intent?.getStringExtra(BackendServiceConst.EXTRA_BORDER))
        return this
    }

    // override
    
    fun userSession(): String? {
        return userSession
    }
    //override
    fun userSession(userSession: String?) {
        this.userSession = userSession
    }

    fun userSuid(): String? {
        return userSuid
    }
    //override
    fun userSuid(userSuid: String?) {
        this.userSuid = userSuid
    }

    // override
    fun userName(): String? {
        return userName
    }

    // override
    fun userName(userName: String?) {
        this.userName = userName
    }

    // override
    fun fair(): String? {
        return fair
    }

    // override
    fun fair(fair: String?) {
        this.fair = fair
    }

    // override
    fun border(): String? {
        return border
    }

    // override
    fun border(border: String?) {
        this.border = border
    }

    companion object {
        fun setArgToIntent(intent: Intent) {
            BaseArg.setArgToIntent(intent)
            intent.putExtra(BackendServiceConst.EXTRA_USER_SESSION, ConfigPref.userSession)
            intent.putExtra(BackendServiceConst.EXTRA_USER_SUID, ConfigPref.userSuid)
            intent.putExtra(BackendServiceConst.EXTRA_USER_NAME, ConfigPref.userName)
            if (null != ConfigPrefHelper.getUsersBorder()) {
                intent.putExtra(BackendServiceConst.EXTRA_FAIR, ConfigPrefHelper.getUsersBorder()?.fair)
                intent.putExtra(BackendServiceConst.EXTRA_BORDER, ConfigPrefHelper.getUsersBorder()?.border)
            } else {
                intent.putExtra(BackendServiceConst.EXTRA_FAIR, "") // fair
                intent.putExtra(BackendServiceConst.EXTRA_BORDER, "") //border
            }
        }

        fun fromPreferences(): CommonArg {
            val commonArg = CommonArg(intent = null)
            val lang = ConfigPref.currentLanguage
            val commKey = ConfigPref.commKey
            val deviceSuid = ConfigPref.deviceSuid
            val deviceType = ConfigPref.deviceType
            val userSession = ConfigPref.userSession
            val userSuid = ConfigPref.userSuid
            val userName = ConfigPref.userName
            val fair = ConfigPrefHelper.getUsersBorder()?.fair
            val border = ConfigPrefHelper.getUsersBorder()?.border
            commonArg.lang = lang
            commonArg.commKey = commKey
            commonArg.deviceSuid = deviceSuid
            commonArg.deviceType = deviceType
            commonArg.userSession = userSession
            commonArg.userSuid = userSuid
            commonArg.userName = userName
            commonArg.fair = fair
            commonArg.border = border
            return commonArg
        }
    }
}